tools/scripts/track-pr

```diff
@@ -195,9 +195,9 @@ class Actions:
             cherry_pick_pr(pr_id)
         else:
             title = gql.get_pr_title(id=pr_id)["node"]["title"]
-            if re.match("(feat|fix):", title) is not None:
+            if re.match("(feat|fix)[^ ]*:", title, re.IGNORECASE) is not None:
```

Comment:
I don't understand the "zero or more instances of any character except space" match between feat/fix and the colon here.  What's an example where that's what's intended?  I'd expect that what you want is a word boundary on either side of feat and fix, and then optional non-word characters separating that from colon.

Also probably should use raw strings for regexps in general.

Perhaps something like this?  I'm not sure if `\W` or `\s` is the more appopriate character class; i'd expect just whitespace, but I'm open to the idea that people are using underscores or something.  I went with `\W` since that's the definition of `\b` anyway (the point between a `\w` and `\W`).  Either way, that asterisk was replaced with `*?` to make it non-greedy - we only want to match up to the first colon.  Doesn't make much difference here, but maybe later if that match is stored?  I dunno.
```suggestion
            if re.match(r"\b(feat|fix)\b\W*?:", title, re.IGNORECASE) is not None:
```

Reply:
There can optionally be a component after the type, something like `feat(api): add stuff`, so `\W` is specifically not what I want. That part could be more specific than `[^ ]*`, but I figured I'd err on the side of simplicity and inclusion (on the basis that the consequence of a false match is that someone has to take a moment to mark the item as done, while the consequence of a false non-match is that the change is completely left out of the testing process). A leading `\b` would be redundant, since `re.match` only matches from the start of the string. A following one would serve more function, though leaving it out again improves simplicity and inclusion, so I think I prefer not to have it. If we ever use any capture groups from this match, we can add non-greediness then.

I'll make the literals raw and change `[^ ]` to `\S`.

Reply:
I probably would've just gone with `(\(\w+\))?`, but to each his own. :stuck_out_tongue:  False matching does sound like the right failure mode.